ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -52,62 +75,117 @@ static void propagate_layout(const std::shared_ptr<Node>& src, const std::shared
-                constant->set_friendly_name(node->get_friendly_name() + "/scale/Divide_Factor");
+                constant->set_friendly_name(nodes[0]->get_friendly_name() + "/scale/Divide_Factor");
 
-                auto new_op = std::make_shared<op::v1::Divide>(node, constant);
-                new_op->set_friendly_name(node->get_friendly_name() + "/scale/Divide");
-                propagate_layout(node, new_op);
+                auto new_op = std::make_shared<op::v1::Divide>(nodes[0], constant);
+                new_op->set_friendly_name(nodes[0]->get_friendly_name() + "/scale/Divide");
+                propagate_layout(nodes[0], new_op);
                 return new_op;
             },
             false));
     }
 
     void add_mean_impl(const std::vector<float>& values) {
         m_actions.emplace_back(std::make_tuple(
-            [values](const std::shared_ptr<Node>& node) {
+            [values](const std::vector<std::shared_ptr<Node>>& nodes, PreprocessingContext&) {
+                OPENVINO_ASSERT(!nodes.empty(), "Internal error: Can't apply mean preprocessing for empty input.");
+                OPENVINO_ASSERT(nodes.size() == 1,
+                                "Can't apply scale preprocessing for multi-plane input. Suggesting to convert current "
+                                "image to RGB/BGR color format using 'convert_color'");
                 Shape shape;
                 if (values.size() == 1) {
                     shape = Shape{1};
                 } else {
-                    shape = construct_mean_scale_shape(node, values.size());
+                    shape = construct_mean_scale_shape(nodes[0], values.size());
                 }
                 auto constant = op::v0::Constant::create(element::f32, shape, values);
-                constant->set_friendly_name(node->get_friendly_name() + "/mean/Mean_Const");
+                constant->set_friendly_name(nodes[0]->get_friendly_name() + "/mean/Mean_Const");
 
-                auto new_op = std::make_shared<op::v1::Subtract>(node, constant);
-                new_op->set_friendly_name(node->get_friendly_name() + "/mean/Subtract");
-                propagate_layout(node, new_op);
+                auto new_op = std::make_shared<op::v1::Subtract>(nodes[0], constant);
+                new_op->set_friendly_name(nodes[0]->get_friendly_name() + "/mean/Subtract");
+                propagate_layout(nodes[0], new_op);
                 return new_op;
             },
             false));
     }
 
     void add_convert_impl(const element::Type& type) {
         m_actions.emplace_back(std::make_tuple(
-            [type](const std::shared_ptr<Node>& node) {
-                if (node->get_element_type().is_dynamic()) {
+            [type](const std::vector<std::shared_ptr<Node>>& nodes, PreprocessingContext&) {
+                OPENVINO_ASSERT(!nodes.empty(), "Internal error: Can't set element type for empty input.");
+                OPENVINO_ASSERT(nodes.size() == 1,
```

Comment:
logically, we can allow convert element type for multi plane cases..

Reply:
Done, now convert_element_type allows multiplane color format